pkg/appmanager/resourceConfig.go

```diff
@@ -931,14 +931,56 @@ func (rc *ResourceConfig) SetMonitor(pool *Pool, monitor Monitor) {
 
 	if !found {
 		pool.MonitorNames = append(pool.MonitorNames, toFind)
+		updated = true
 	}
 	for i, mon := range rc.Monitors {
 		if mon.Name == monitor.Name && mon.Partition == monitor.Partition {
-			rc.Monitors[i] = monitor
-			return
+			if !reflect.DeepEqual(rc.Monitors[i], monitor) {
+				rc.Monitors[i] = monitor
+				updated = true
+			}
+			return updated
 		}
 	}
 	rc.Monitors = append(rc.Monitors, monitor)
+	return updated
+}
+
+func (rc *ResourceConfig) RemoveMonitor(pool, monitor string) bool {
+	var removed bool
+	for i, pl := range rc.Pools {
+		if pl.Name == pool {
+			for j, mon := range pl.MonitorNames {
+				if mon == monitor {
+					if i >= len(pl.MonitorNames)-1 {
```

Comment:
I don't understand the use of 'i' in this context, and why you wouldn't always do the 'else' case here.

Reply:
That should be a 'j'. If we are looking at the last monitor in the list, we can't access j+1 in the 'else' case.

Reply:
Ah, ok.